en/sensor/rangefinders.md

```diff
@@ -0,0 +1,171 @@
+
+Wiring and other information for the I2C variants can be found in the topic [Lightware SFxx Lidar](../sensor/sfxx_lidar.md).
+
+> **Tip** Not all devices are supported for both serial and I2C; check the parameters for more information and supported models.
+
+
+### TeraRanger One
+
+TeraRanger provide a number of lightweight distance measurement sensor based on infrared Time-of-Flight (ToF) technology. They are typically faster and have greater range than sonar, and smaller and lighter than laser-based systems. PX4 supports:
+* [TeraRanger One](http://www.teraranger.com/products/teraranger-one/) (0.2 - 14 m)
+* [TeraRanger Evo](https://www.terabee.com/portfolio-item/teraranger-evo-infrared-distance-sensor/) (0.5 – 60 m)
+
+These rangefinders must be connected via the I2C bus (using an [I2C adapter](http://www.teraranger.com/product/teraranger-i2c-adapter/)).
+
+The sensors are enabled using the parameter [SENS_EN_TRANGER](../advanced_config/parameter_reference.md#SENS_EN_TRANGER) (you can set the type of sensor or that PX4 should auto-detect the type).
+
+> **Info** The *Terranger One* is used in the [Qualcomm Snapdragon Flight](../flight_controller/snapdragon_flight.md). 
+
+### uLanding Radar
+
+The *Aerotenna* [uLanding](https://aerotenna.com/sensors/#ulanding) altimeter is compact microwave rangefinder that has been optimised for use on UAVs. It has a sensing range of 45m. A particular advantages of this product are that it can operate effectively in all weather conditions and over all terrain types (including water).
+
+The uLanding Radar is not present in "most" firmware by default and must be started by updating a configuration file (rather than by a parameter). More information can be found here: [uLanding Radar](../sensor/ulanding_radar.md).
+
+
+### Other
+
+PX4 also supports the Bebop rangefinder.
+
+
+## Rangefinder Configuration
+
+The rangefinder is configured using [EKF2\_RNG\_*](../advanced_config/parameter_reference.md#EKF2_RNG_POS_X) parameters, e.g. delay, offset of rangefinder from vehicle body, etc.
+
+The rangefinder can also be used to improve altitude measurements even if it is not the primary height source by setting [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_RNG_AID) to 1. Other parameters affecting "Range Aid" are prefixed with `EKF2\_RNG\_A\_`.
+
+
+## Testing
+
+The easiest way to test the rangefinder is to vary the range and compare to the values detected by PX4. The sections below show some approaches to getting the measured range.
+
+### QGC MAVLink Inspector
+
+The *QGroundControl MAVLink Inspector* tool lets you check the altitude sent from the vehicle. If you set the vehicle to use a rangefinder as the main height source, then this should reflect the sensor values:
+
+1. Set the rangefinder as the main height source (TBD)

```

Comment:
@bkueng I think this PR is ready for review. The only outstanding bit AFAIK is this setting.